paimon-core/src/main/java/org/apache/paimon/iceberg/IcebergMigrator.java

```diff
@@ -0,0 +1,344 @@
+                        Thread.sleep(100);
+                    }
+                }
+                // roll back all renamed path
+                for (Map.Entry<Path, Path> entry : rollback.entrySet()) {
+                    Path newPath = entry.getKey();
+                    Path origin = entry.getValue();
+                    if (paimonFileIO.exists(newPath)) {
+                        paimonFileIO.rename(newPath, origin);
+                    }
+                }
+
+                throw new RuntimeException("Migrating failed because exception happens", e);
+            }
+            try (BatchTableCommit commit = paimonTable.newBatchWriteBuilder().newCommit()) {
+                commit.commit(new ArrayList<>(commitMessages));
+                LOG.info("paimon commit success! Iceberg data files has been migrated to paimon.");
+            }
+        } catch (Exception e) {
+            paimonCatalog.dropTable(paimonIdentifier, true);
+            throw new RuntimeException("Migrating failed", e);
+        }
+    }
+
+    @Override
+    public void deleteOriginTable(boolean delete) throws Exception {}
+
+    @Override
+    public void renameTable(boolean ignoreIfNotExists) throws Exception {}
+
+    public int getIcebergNewestSnapshotId() {
+        Path versionHintPath =
+                new Path(icebergPathFactory.metadataDirectory(), VERSION_HINT_FILENAME);
+        try {
+            return Integer.parseInt(paimonFileIO.readFileUtf8(versionHintPath));
+        } catch (IOException e) {
+            throw new RuntimeException(
+                    "read iceberg version-hint.text failed. Iceberg metadata path: "
+                            + icebergPathFactory.metadataDirectory());
+        }
+    }
+
+    public Schema icebergSchemaToPaimonSchema(IcebergMetadata icebergMetadata) {
+        // get iceberg current schema
+        IcebergSchema icebergSchema =
+                icebergMetadata.schemas().get(icebergMetadata.currentSchemaId());
+
+        // get iceberg current partition spec
+        int currentPartitionSpecId = icebergMetadata.defaultSpecId();
+        IcebergPartitionSpec currentIcebergPartitionSpec =
+                icebergMetadata.partitionSpecs().get(currentPartitionSpecId);
+
+        List<DataField> dataFields =
+                icebergSchema.fields().stream()
+                        .map(IcebergDataField::toDatafield)
+                        .collect(Collectors.toList());
+
+        List<String> partitionKeys =
+                currentIcebergPartitionSpec.fields().stream()
+                        .map(IcebergPartitionField::name)
+                        .collect(Collectors.toList());
+
+        return new Schema(
+                dataFields, partitionKeys, Collections.emptyList(), Collections.emptyMap(), null);
+    }
+
+    private List<IcebergManifestFileMeta> checkAndFilterManifestFiles(
+            List<IcebergManifestFileMeta> icebergManifestFileMetas) {
+        if (!ignoreDelete) {
```

Comment:
Why do we need such an option? If there are deletion vectors in Iceberg, and user uses this option, then the resulting data will be incorrect. Incorrect data are useless to the users.